kernel/reduction.ml

```diff
@@ -479,7 +479,13 @@ and eqappr cv_pb l2r infos (lft1,st1) (lft2,st2) cuniv =
     | (FFlex fl1, c2)      ->
        (match unfold_reference infos fl1 with
 	| Some def1 ->
-           eqappr cv_pb l2r infos (lft1, (def1, v1)) appr2 cuniv
+          (** By virtue of the previous case analyses, we know [c2] is rigid.
+              Conversion check to rigid terms eventually implies full weak-head
+              reduction, so instead of repeatedly performing small-step
+              unfoldings, we perform reduction with all flags on. *)
+            let all = RedFlags.red_add_transparent all (RedFlags.red_transparent (info_flags infos)) in
+            let r1 = whd_stack (infos_with_reds infos all) def1 v1 in
```

Comment:
`(infos_with_res infos (red_add_transparent all (red_transparent (info_flags infos))))` is just `infos` when `infos` has betaiotazeta, which it always does since we're always called from 
```coq
let clos_gen_conv trans cv_pb l2r evars env univs t1 t2 =
  let reds = CClosure.RedFlags.red_add_transparent betaiotazeta trans in
  let infos = create_clos_infos ~evars reds env in
  ccnv cv_pb l2r infos el_id el_id (inject t1) (inject t2) univs
```

Reply:
WDYM? `all` has more flags set on than `betaiotazeta`, most notably δ.